net_handler/tcp_conn.c

```diff
@@ -152,14 +149,19 @@ static int recv_new_msg (int conn_fd, int challenge_fd)
 
 	int err = parse_msg(conn_fd, &msg_type, &len_pb, &buf);
 	if (err == 0) { // Means there is EOF while reading which means client disconnected
-		// log_printf(DEBUG, "EOF occurred");
 		return -1; 
 	}
 	else if (err == -1) { // Means there is some other error while reading
 		return -2; 
 	}
+	
 	//unpack according to message
 	if (msg_type == CHALLENGE_DATA_MSG) {
+		//socket address structure for the UNIX socket to executor for challenge data
+		struct sockaddr_un exec_addr;
+		exec_addr.sun_family = AF_UNIX;
+		strcpy(exec_addr.sun_path, CHALLENGE_SOCKET);
```

Comment:
I think you should use the struct initializer I had earlier, or use memset

Reply:
This generated errors when I compiled it on my Mac (saying you can't initialize a`char *` with non-constant array or something like that). I looked it up online and it seems that it's good practice to do it this way (i.e. use `strcpy`. This way also makes it clear what the names of the fields that we're initializing are, as opposed to `struct sockaddr_un exec_addr = { AF_UNIX, CHALLENGE_SOCKET }` which doesn't tell you what the names of the fields are (`sun_family` and `sun_path`)

Reply:
you should still use memset to set rest of the elements to 0 I believe. I know thats good practice for sockaddr_in atleast.

Reply:
Fixed it